torch/_inductor/codegen/cuda/cuda_kernel.py

```diff
@@ -265,21 +263,196 @@ def row_or_column_stride(self, node: IRNode, default_value: int = 0) -> str:
             )
 
 
+class CUDATemplateBuffer(TemplateBuffer):
+    def __init__(
+        self,
+        template: "CUDATemplate",  # type: ignore[name-defined]
+        op: "cutlass_gemm_op.GemmOperation",  # type: ignore[name-defined]
+        epilogue_nodes: Optional[
+            List[IRNode]
+        ] = None,  # We need a new instance of this op every time we fuse
+        workspace_size: Optional[
+            Union[int, Callable[[], int]]
+        ] = None,  # May be an int or a callback returning an int
+        merged_input_nodes: Optional[
+            List[IRNode]
+        ] = None,  # input nodes of the template + epilogue nodes
+        name: Optional[
+            str
+        ] = None,  # Name of the buffer, must be registered with the graph if passed.
+        **render_kwargs,  # passed through to template_node.render
+    ):
+        if epilogue_nodes is None:
+            epilogue_nodes = []
+        if merged_input_nodes is None:
+            assert (
+                epilogue_nodes is None or len(epilogue_nodes) == 0
+            ), "If epilogue nodes are passed, merged_input_nodes must be passed as well"
+            input_nodes = template.input_nodes
+        else:
+            input_nodes = merged_input_nodes
+        super().__init__(
+            template.layout, input_nodes, self._make_kernel_render, name=name
+        )
+        self.template = template
+        self.op = op
+        self._epilogue_nodes = epilogue_nodes
+        # TODO: Once we support non-pointwise fusions, layout might be modified by epilogues
+        self.layout = template.layout
+        self._render_kwargs = render_kwargs
+        # Global memory (in bytes) needed for this template.
+        self._workspace_size = workspace_size
+
+    @property
+    def epilogue_nodes(self):
+        # read-only property to signal it should not be mutated
+        return self._epilogue_nodes
+
+    @property
+    def workspace_size(self):
+        return self.get_workspace_size()
+
+    def get_workspace_size(self):
+        if callable(self._workspace_size):
```

Comment:
The workspace size is not necessarily known yet when this instance is being created, so it can also be passed as a callback that may be evaluated lazily.